atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -560,44 +563,52 @@ private void initPoolSslOracle(Properties props) throws Exception {
             }
 
             // server_dn_matching
-            String sMatchServerDN = dbConfig.getConnectionParameter(DBConfigConnectionParameter.MATCH_SERVER_DN);
-            if (sMatchServerDN != null && sMatchServerDN.length() > 0) {
+            String sMatchServerDN = dbConfig.getMatchServerDn();
+            if (!Strings.isNullOrEmpty(sMatchServerDN)) {
                 props.setProperty("oracle.net.ssl_server_dn_match", "true");
                 log.info("Will require the server certificate DB to match: " + sMatchServerDN);
 
                 // modify the config to have a URL suffix
-                dbConfig.setUrlSuffix(DatabaseConstants.DB_ORACLE_SECURITY_SUFFIX);
+                dbConfig = ImmutableConnectionConfig.builder().from(dbConfig)
+                        .urlSuffix(DatabaseConstants.DB_ORACLE_SECURITY_SUFFIX)
+                        .build();
             } else {
                 // set the closer if we don't have a dn to match
-                dbConfig.setUrlSuffix(DatabaseConstants.DB_ORACLE_NO_SECURITY_SUFFIX);
+                dbConfig = ImmutableConnectionConfig.builder().from(dbConfig)
+                        .urlSuffix(DatabaseConstants.DB_ORACLE_NO_SECURITY_SUFFIX)
+                        .build();
             }
 
             /*
              * Enable client SSL certificate support. "two-way" SSL in Oracle parlance.
              */
             if (dbConfig.getTwoWaySsl()) {
-                props.setProperty("javax.net.ssl.keyStore", keystoreManager.getKeystorePath());
-                props.setProperty("javax.net.ssl.keyStorePassword", keystoreManager.getKeystorePassword());
+                Preconditions.checkArgument(dbConfig.getKeystorePath().isPresent());
+                Preconditions.checkArgument(dbConfig.getKeystorePassword().isPresent());
+                props.setProperty("javax.net.ssl.keyStore", dbConfig.getKeystorePath().get());
+                props.setProperty("javax.net.ssl.keyStorePassword", dbConfig.getKeystorePassword().get());
             }
         } else {
             /*
              * Set no security suffix if security isn't enabled.
              */
-            dbConfig.setUrlSuffix(DatabaseConstants.DB_ORACLE_NO_SECURITY_SUFFIX);
+            dbConfig = ImmutableConnectionConfig.builder().from(dbConfig)
+                    .urlSuffix(DatabaseConstants.DB_ORACLE_NO_SECURITY_SUFFIX)
+                    .build();
 
             /*
              * Ensure a PROTOCOL is set.  Default to "tcp" if none is set.
              */
             if (sProtocol == null) {
-                dbConfig.addConnectionParameter(
-                        DBConfigConnectionParameter.PROTOCOL,
-                        DatabaseConstants.DEFAULT_DB_CONNECTION_PARAM_PROTOCOL);
+                dbConfig = ImmutableConnectionConfig.builder().from(dbConfig)
+                        .protocol(ConnectionProtocol.TCP)
```

Comment:
Seems like we would still want this to reference a 'default' constant somewhere instead of having the default covertly specified here as TCP.